docs/contracts/constant-state-variables.rst

```diff
@@ -12,6 +12,14 @@ for ``immutable``, it can still be assigned at construction time.
 The compiler does not reserve a storage slot for these variables, and every occurrence is
 replaced by the respective value.
 
+Compared to regular state variables, the gas costs of constant and immutable variables
+are much lower. For a constant variables, the expression assigned to it is copied to
+all the places where it is accessed and also re-evaluated each time. This allows for local
+optimizations. Immutable variables are evaluated once at construction time and their value
+is copied to all the places where they are accessed. For this, 32 bytes are reserved
```

Comment:
```suggestion
is copied to all the places where they are accessed. 32 bytes are reserved
```

Reply:
I try to avoid starting a sentence with a number, so I did a different change.